spec/rails_integration_spec.rb

```diff
@@ -106,14 +106,14 @@ def prepare_params(params)
   end
 
   describe "nested_array" do
-    it "responds with a 400 when the nested array is not supplied properly" do
-      params = {
-        'filter' => 'state'
-      }
+    it "responds with a 400 when the nested array is provided as nil" do
+      expect { get :nested_array, **prepare_params({ filter: { state: nil } }) }
+        .to raise_error(RailsParam::InvalidParameterError)
```

Comment:
This shouldn't be the case, should it? An array should be able to be nil, nested or otherwise. I believe the expectations should be:
- If param is missing/nil and is required, it raises an error
- if param is missing/nil and not required, it stays nil
- if param is missing/nil, not required, and defaults to [], then it's coerced to []

So I believe we should be expecting this to not raise an error. Am I missing something?

Reply:
@ifellinaholeonce sorry for the confusion, this is currently in a bit of a chaos after a few hours of discussion with @iMacTia about how to solve this. The gist is. I implemented those tests first as a 200. They failed. I have applied the change in param_evaluator.rb. They started passing. But the original test started failing. Anyway. I will make them pass now, and I will move to another thread here to discuss that original test.

Reply:
This test should fail, but only because of the nested validation on array items. If that was not there, then this would be a valid payload